src/main/java/com/newestuser/imen/bgdate/BulgarianDate.java

```diff
@@ -0,0 +1,174 @@
+        put(6, new DayMonthRange(1, 21, "Шехтем", "Сетем"));
+        put(7, new DayMonthRange(1, 22, "Сетем", "Бехти"));
+        put(8, new DayMonthRange(1, 22, "Бехти", "Нунтем"));
+        put(9, new DayMonthRange(1, 20, "Нунтем", "Елем"));
+        put(10, new DayMonthRange(1, 21, "Елем", "Ениалем"));
+        put(11, new DayMonthRange(1, 20, "Ениалем", "Алтом"));
+        put(12, new DayMonthRange(1, 20, "Алтом", "Алем"));
+    }};
+
+    private static Map<Integer, String> YEAR_TYPES = new HashMap<>() {{
+        put(0, "Верени");
+        put(1, "Дилом");
+        put(2, "Имен");
+        put(3, "Теку");
+        put(4, "Бечин");
+        put(5, "Тох");
+        put(6, "Етх");
+        put(7, "Докс");
+        put(8, "Сомор");
+        put(9, "Шегор");
+        put(10, "Барс");
+        put(11, "Дван");
+    }};
+
+    private static Map<String, Integer[]> DAY_CONVERSION_VALUES = new HashMap<>() {{
+        put("Алем", new Integer[]{9, 22});
+        put("Тутом", new Integer[]{9, 21});
+        put("Читем", new Integer[]{8, 22});
+        put("Твирем", new Integer[]{9, 22});
+        put("Вечем", new Integer[]{8, 22});
+        put("Шехтем", new Integer[]{9, 21});
+        put("Сетем", new Integer[]{9, 22});
+        put("Бехти", new Integer[]{9, 21});
+        put("Нунтем", new Integer[]{8, 20});
+        put("Елем", new Integer[]{10, 21});
+        put("Ениалем", new Integer[]{10, 20});
+        put("Алтом", new Integer[]{10, 21});
+    }};
+
+    /**
+     * Converts the passed gregorian day of the month to the day according to the Bulgarian calendar.
+     *
+     * @param currentDay current day in a given gregorian month.
+     * @param bgMonth    current month according to the Bulgarian calendar.
+     * @param month      current month according to the Gregorian calendar.
+     * @return the converted Bulgarian day.
+     */
+    private static int findBulgarianDay(int currentDay, String bgMonth, int month) {
+        DayMonthRange dayMonthRange = GREGORIAN_MONTH_TO_BG_MONTH.get(month);
+        bgMonth = dayMonthRange.getBulgarianMonth(currentDay);
+        int bulgarianDateDay = currentDay;
+        if (DAY_CONVERSION_VALUES.containsKey(bgMonth)) {
+            if (dayMonthRange.findIsInFirstMonthPart(currentDay)) {
+                bulgarianDateDay += DAY_CONVERSION_VALUES.get(bgMonth)[0];
+            } else {
+                bulgarianDateDay -= DAY_CONVERSION_VALUES.get(bgMonth)[1];
+            }
+        }
+        if ((month > 2) && !findIsLeapYear(Year.now().getValue())) {
+            bulgarianDateDay -= 1;
+        }
+        return bulgarianDateDay;
+    }
+
+    /**
+     * Converts the passed gregorian month to the month according to the Bulgarian calendar.
+     *
+     * @param month      current month according to the Gregorian calendar.
+     * @param currentDay current day in a given gregorian month.
+     * @return the converted Bulgarian month.
+     */
+
```

Comment:
make sure that you do not have floating documentation here and on the other methods as well